### ISSUE TITLE
Dynamically changing selectable.

### ISSUE THREAD

**Issue by gillez** [olifolkerd/tabulator#1610]
Hi,

Is it possible to dynamically change the selectable parameter on an existing table?  I want to add a button to switch between single-select and multi-select.  At the moment I am calling "new Tabulator()" again to reinitialise the table when switching between the modes.  This works but redraws the table and so I lose the current scroll position, column width changes etc.

Is it easy to add a workaround for this?

Thanks,
Giles

**Comment by Rodbjartson** [comment]
Maybe you can use the selectableCheck callback?
From the docs:
```
selectableCheck:function(row){
        //row - row component
        return row.getData().age > 18; //allow selection of rows where the age is greater than 18
    },
```

Make your table multi-selectable, then you can add a check whether the button is checked or not. If it does allow multi, then return true, else false if there is already one selected row

**Comment by gillez** [comment]
Thanks for the suggestion.  This doesn't quite do what I want though.  If my multi checkbox is turned off, I want to allow the new row but turn off the selection on the old row, rather than disallow the new row.  This is the same behaviour as setting selectable=1.

The only solution I have found is to set selectable="highlight" and handling the selection logic myself.  I have now implemented this.

**Comment by olifolkerd** [comment]
Hey @gillez 

in this case i would suggest that you disable row selection:
```js
selectable:false,
```

and implement it using the ***rowClick*** callback to toggle row selection, you could then reference the current state of your toggle button to determine what to do with the selection:

```js
var table = new Tabulator("#example-table", {
    selectable:false,
    rowClick:function(e, row){
        //e - the click event object
        //row - row component

       //add some logic to determine if selection is possible then toggle selection

      row.toggleSelect();
    },
});
```

the row component has a variety of function that can be used to manipulate selection, full details can be found in the [Row Component Documentation](http://tabulator.info/docs/4.1/components#component-row)

 I hope that helps,

Cheers

Oli :)

**Comment by gillez** [comment]
Hi Oli,

Thanks for the suggestion.  I did try setting selectable to false and handling the selection myself by calling table.selectRow.  However, this causes an error:

```
tabulator.min.js:9 Uncaught TypeError: Cannot set property 'selected' of undefined
    at G._selectRow (tabulator.min.js:9)
    at G.selectRows (tabulator.min.js:9)
    at t.c.selectRow (tabulator.min.js:4)
    at Object.m [as rowClick] (app.js:8)
    at HTMLDivElement.<anonymous> (tabulator.min.js:3)
```

I get a similar error if I call row.toggleSelect.  It sems that the row.selected property is not set with selectable=false.

Setting selectable="highlight" fixes this.  (I don't personally really like the highlighting so have removed this in the CSS file, so the behaviour looks the same as selectable off).

Tabulator is great by the way.  Thanks for all your work in creating this.  I love that it is so customisable and that there are so many hooks, so if something doesn't work the way I want it to, it's easy to add a hook and implement the behaviour that I want myself.

Cheers,
Giles

**Comment by olifolkerd** [comment]
Hey @gillez 

If you are still having issues with this, would you mind creating a js fiddle or code pen that replicates the issue as i cant seem to replicate the error you are getting on my end.

Cheers

Oli :)

**Comment by gillez** [comment]
Hi Oli,

Thanks for the reply.  I'll try to replicate this in a JS Fiddle when I get the chance.  I'll post a link when it's done.

Thanks,
Giles

**Comment by gillez** [comment]
Hi Oli,
I've just noticed that you have closed this.  Apologies for not making the JS Fiddle.  Things were a bit busy at the time and I never got around to it.  There still busy now (when are things not busy) but I'll see if the issue is still there today and will create a fiddle if it is.
Cheers,
Giles

**Comment by olifolkerd** [comment]
The issue will be resolved in todays 4.2.1 release
